plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -208,7 +211,7 @@ public InstanceKey saveToSnapshot(String owner) throws MachineException {
 
             docker.removeImage(registry + '/' + repository, false);
 
-            return new DockerInstanceKey(repository, "latest", registry, digest);
+            return new DockerMachineSource(repository).setRegistry(registry).setDigest(digest).setTag(LATEST_TAG);
```

Comment:
We don't use fluent style setters, please consider usage of with methods instead